lib/pdk/util/bundler.rb

```diff
@@ -0,0 +1,129 @@
+require 'bundler'
+require 'tty-spinner'
+require 'pdk/util'
+require 'pdk/cli/exec'
+
+module PDK
+  module Util
+    module Bundler
+      class BundleHelper; end
+
+      def self.ensure_bundle!
+        bundle = BundleHelper.new
+
+        if bundle.has_gemfile?
+          if !bundle.locked?
+            bundle.lock
+          end
+
+          if !bundle.installed?
+            bundle.install
+          end
+        end
+      end
+
+      class BundleHelper
+        def has_gemfile?
+          # return a pure boolean
+          !!gemfile
+        end
+
+        def locked?
+          # return a pure boolean
+          !!gemfile_lock
+        end
+
+        def installed?
+          output_start(_("Checking for missing Gemfile dependencies"))
+
+          result = invoke('check', "--gemfile=#{gemfile}", "--path=#{pdk_cachedir}")
```

Comment:
Yeah, right now it points to a generic directory we can use for any sort of caching we need to do with PDK and the default gem pathing of `ruby/2.1.0/...` gets added to the passed in value by bundler, so the value that is getting interpolated there isn't really specific to gems.